### PR TITLE
IAS attestation report verification by enclave manager

### PR DESCRIPTION
**Pull request by manju956**
1. Changes to verify quote status, attestation report signature, mr_enclave and report data are included.
2. Commit history corresponding to verify_ias_report module from PDO is included.

### CODE REVIEW

File: examples/enclave_manager/tcf_enclave_manager/tcf_enclave_bridge.py [hunk 1]
```diff
@@ -222,16 +233,21 @@ def create_signup_info(originator_public_key_hash, nonce):
...
             })
-
         # Grab the EPID pseudonym and put it in the enclave-persistent ID for the
         # signup info
         verification_report_dict = json.loads(response['verification_report'])
         signup_info['enclave_persistent_id'] = verification_report_dict.get('epidPseudonym')
 
+        mr_enclave = get_enclave_measurement()
+        status = verify_enclave_info(json.dumps(signup_info), mr_enclave, originator_public_key_hash)
+        logger.info("status - %d", status)
```
Comment (manojgop) [inline examples/enclave_manager/tcf_enclave_manager/tcf_enclave_bridge.py]: remove this log

File: tc/sgx/trusted_worker_manager/enclave_wrapper/signup.cpp [hunk 2]
```diff
@@ -235,3 +235,46 @@ tcf_err_t tcf::enclave_api::enclave_data::UnsealEnclaveData(
...
+tcf_err_t tcf::enclave_api::enclave_data::VerifyEnclaveInfo(
+    const std::string& enclaveInfo,
+    const std::string& mr_enclave,
+    const std::string& originator_public_key_hash) {
+    tcf_err_t result = TCF_SUCCESS;
+    try {
+        // xxxxx call the enclave
+        sgx_enclave_id_t enclaveid = g_Enclave[0].GetEnclaveId();
+        tcf_err_t presult = TCF_SUCCESS;
+        printf("singup::VerifyEnclaveInfo\n");
```
Comment (manojgop) [inline tc/sgx/trusted_worker_manager/enclave_wrapper/signup.cpp]: remove printf

File: tc/sgx/trusted_worker_manager/enclave_wrapper/signup.cpp [hunk 3]
```diff
@@ -235,3 +235,46 @@ tcf_err_t tcf::enclave_api::enclave_data::UnsealEnclaveData(
...
+        sgx_status_t sresult = g_Enclave[0].CallSgx(
+            [ enclaveid,
+              &presult,
+	      enclaveInfo,
```
Comment (manojgop) [inline tc/sgx/trusted_worker_manager/enclave_wrapper/signup.cpp]: correct indentation

File: tc/sgx/trusted_worker_manager/enclave_wrapper/signup_info.cpp [hunk 4]
```diff
@@ -143,7 +145,8 @@ std::map<std::string, std::string> CreateEnclaveData(
     const std::string& originator_public_key_hash
     ) {
     tcf_err_t presult;
-
+    printf("CreateEnclaveData\n");
+    fflush(stdin);
```
Comment (manojgop) [inline tc/sgx/trusted_worker_manager/enclave_wrapper/signup_info.cpp]: remove printf and fflush

File: examples/enclave_manager/tcf_enclave_manager/tcf_enclave_bridge.py (hunk 1 above)
Comment (manju956) [reply examples/enclave_manager/tcf_enclave_manager/tcf_enclave_bridge.py]: Done

File: examples/enclave_manager/tcf_enclave_manager/tcf_enclave_helper.py [hunk 5]
```diff
@@ -67,8 +68,13 @@ def create_enclave_signup_data(cls, tcf_instance_keys = None) :
         enclave_info['enclave_id'] = enclave_data.verifying_key
         enclave_info['proof_data'] = ''
         if not tcf_enclave.enclave.is_sgx_simulator() :
-            enclave_info['proof_data'] = enclave_data.proof_data
+            logger.info("proof data type is: %s", type(enclave_data.proof_data))
```
[the remark this answers is not in the document]
Comment (manju956) [reply examples/enclave_manager/tcf_enclave_manager/tcf_enclave_helper.py]: Done

File: examples/enclave_manager/tcf_enclave_manager/tcf_enclave_helper.py [hunk 6]
```diff
@@ -67,8 +68,13 @@ def create_enclave_signup_data(cls, tcf_instance_keys = None) :
...
+            enclave_info['proof_data'] = str(enclave_data.proof_data)
 
+        # Verify enclave info
+        #logger.info("verifying IAS report")
+        #verify_status = tcf_enclave.verify_enclave_info(enclave_info)
+        #logger.info("verification of IAS report status: %s", verify_status)
```
[the remark this answers is not in the document]
Comment (manju956) [reply examples/enclave_manager/tcf_enclave_manager/tcf_enclave_helper.py]: Done

File: tc/sgx/common/CMakeLists.txt [hunk 7]
```diff
@@ -23,12 +23,25 @@ INCLUDE(CMakeVariables.txt)
...
-SET(COMMON_PRIVATE_INCLUDE_DIRS "." "tests" "crypto" "packages/base64" "packages/parson" "state")
+SET(COMMON_PRIVATE_INCLUDE_DIRS "." "tests" "crypto" "crypto/verify_ias_report" "packages/base64" "packages/parson" "state")
 SET(COMMON_CXX_FLAGS ${DEBUG_FLAGS} "-m64" "-fvisibility=hidden" "-fpie" "-fPIC" "-fstack-protector" "-std=c++11" "-Wall")
 
+################################################################################
+# Generated source
+################################################################################
+
+SET(PROJECT_GENERATED_IAS_SOURCES ${CMAKE_CURRENT_BINARY_DIR}/../crypto/verify_ias_report/ias-certificates.cpp)
+set_source_files_properties(${PROJECT_GENERATED_IAS_SOURCES} PROPERTIES GENERATED TRUE)
```
[the remark this answers is not in the document]
Comment (manju956) [reply tc/sgx/common/CMakeLists.txt]: will remove it

File: tc/sgx/common/CMakeLists.txt [hunk 8]
```diff
@@ -23,12 +23,25 @@ INCLUDE(CMakeVariables.txt)
...
+add_custom_command(OUTPUT  ${PROJECT_GENERATED_IAS_SOURCES}
+                     COMMAND ./build_ias_certificates_cpp.sh
+                     DEPENDS crypto/verify_ias_report/ias-certificates-sim.template crypto/verify_ias_report/ias-certificates-hw.template crypto/verify_ias_report/build_ias_certificates_cpp.sh
+                     WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/../crypto/verify_ias_report
+)
+add_custom_target(generate-ias-files DEPENDS ${PROJECT_GENERATED_IAS_SOURCES})
```
[the remark this answers is not in the document]
Comment (manju956) [reply tc/sgx/common/CMakeLists.txt]: i will remove it

File: tc/sgx/common/crypto/verify_ias_report/ias-certificates-hw.template [hunk 9]
```diff
@@ -0,0 +1,183 @@
+/* Copyright 2018 Intel Corporation
```
[the remark this answers is not in the document]
Comment (manju956) [reply tc/sgx/common/crypto/verify_ias_report/ias-certificates-hw.template]: sure, will remove it

File: tc/sgx/trusted_worker_manager/enclave/signup_enclave.cpp [hunk 10]
```diff
@@ -266,3 +274,153 @@ void CreateSignupReportData(const char* pOriginatorPublicKeyHash,
...
+    tcf::error::ThrowIfNull(proof_data_parsed.value, "Failed to parse the proofData, badly formed JSON");
+    JSON_Object* proof_object = json_value_get_object(proof_data_parsed);
+    tcf::error::ThrowIfNull(proof_object, "Invalid proof, expecting object");
+
+    svalue = json_object_dotget_string(proof_object, "ias_report_signature");
+    tcf::error::ThrowIfNull(svalue, "Invalid proof_signature");
+    const std::string proof_signature(svalue);
+
+    //Parse verification report
+    //const char* verification_report = json_object_dotget_string(proof_object, "verification_report");
```
[the remark this answers is not in the document]
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/signup_enclave.cpp]: Done

File: tc/sgx/trusted_worker_manager/enclave_wrapper/signup.cpp (hunk 2 above)
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave_wrapper/signup.cpp]: Done

File: tc/sgx/trusted_worker_manager/enclave_wrapper/signup.cpp (hunk 3 above)
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave_wrapper/signup.cpp]: Done

File: tc/sgx/trusted_worker_manager/enclave_wrapper/signup_info.cpp (hunk 4 above)
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave_wrapper/signup_info.cpp]: Done

File: tc/sgx/trusted_worker_manager/enclave/signup_enclave.cpp [hunk 11]
```diff
@@ -241,7 +249,7 @@ void CreateSignupReportData(const char* pOriginatorPublicKeyHash,
     //
     // WARNING - WARNING - WARNING - WARNING - WARNING - WARNING - WARNING
     std::string hashString;
-
+
```
Comment (danintel) [inline tc/sgx/trusted_worker_manager/enclave/signup_enclave.cpp]: Remove extra space added on a blank line

Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/signup_enclave.cpp]: Done

File: tc/sgx/common/crypto/verify_ias_report/build_ias_certificates_cpp.sh [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+test -e RK_PUB.zip
+echo "Zipped certificated downloaded"
+
+unzip -o RK_PUB.zip
+test -e AttestationReportSigningCACert.pem
+
+echo ""
+echo -n "Building ias-certificates.cpp ... "
+#replace the placemark in the template with the der certificate
+sed -e '/IAS_REPORT_SIGNING_CA_CERT_PEM_PLACEMARK/ r ./AttestationReportSigningCACert.pem' -e 's/IAS_REPORT_SIGNING_CA_CERT_PEM_PLACEMARK//' < ias-certificates.template > ias-certificates.cpp
```
[the remark this answers is not in the document]
Comment (manju956) [reply tc/sgx/common/crypto/verify_ias_report/build_ias_certificates_cpp.sh]: Done

**Pull request merged**